src/Compilers/Core/Portable/Emit/EditAndContinue/DeltaMetadataWriter.cs

```diff
@@ -1117,51 +1147,77 @@ private void PopulateEncLogTableRows(TableIndex tableIndex, int firstRowId, int
             }
         }
 
-        private void PopulateEncMapTableRows(ImmutableArray<int> rowCounts, List<int> customAttributeEncMapRows, List<int> paramEncMapRows)
+        private void PopulateEncMapTableRows(ImmutableArray<int> rowCounts, ArrayBuilder<int> paramEncMapRows)
         {
             // The EncMap table maps from offset in each table in the delta
             // metadata to token. As such, the EncMap is a concatenated
             // list of all tokens in all tables from the delta sorted by table
             // and, within each table, sorted by row.
             var tokens = ArrayBuilder<EntityHandle>.GetInstance();
             var previousSizes = _previousGeneration.TableSizes;
-            var deltaSizes = this.GetDeltaTableSizes(rowCounts);
+            var deltaSizes = GetDeltaTableSizes(rowCounts);
+
+            // Add tokens in order based on TableIndex. Rows for each table are assumed to have been already ordered.
+            for (var tableIndex = (TableIndex)0; tableIndex <= TableIndex.GenericParamConstraint; tableIndex++)
```

Comment:
>for (var tableIndex = (TableIndex)0; [](http://example.com/codeflow?start=12&length=36)

Is this just a refactoring of the existing code or was there an issue with the previous code?

Reply:
This change allows us to remove the sorting. The various tables were already sorted they just needed to be emitted in the right order.